lib/pat.c

```diff
@@ -1401,6 +1411,28 @@ pat_node_get_key_position(grn_pat *pat, pat_node_common *node)
   }
 }
 
+static inline void
+pat_node_set_key_offset(grn_pat *pat,
+                        pat_node_common *node,
+                        uint64_t key_offset)
+{
+  if (pat_is_key_large(pat)) {
+    node->node_large.key = key_offset;
+  } else {
+    node->node.key = (uint32_t)key_offset;
+  }
+}
+
+static inline uint64_t *
+pat_node_get_key_address(grn_pat *pat, pat_node_common *node)
+{
+  if (pat_is_key_large(pat)) {
+    return &(node->node_large.key);
+  } else {
+    return (uint64_t *)&(node->node.key);
```

Comment:
Will it automatically be casted to `unit64_t`?

Reply:
In my opinion, I should not rely on implicit cast as much as possible.
Because It is difficult whether type expand by intentional or not.